scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Denotation.scala

```diff
@@ -0,0 +1,110 @@
+package scala.meta
+package semantic
+
+import org.scalameta.adt._
+import org.scalameta.data._
+
+@data class Denotation(flags: Long, name: String, info: String) extends HasFlags {
```

Comment:
I still believe that this overloads a term is a very misleading way.

In your case every denotation has a corresponding symbol to start with. The symbol is the original abstraction, it's just that it makes practical sense to serialize\pass unique identifier of the symbol instead of serializing ther entire information about it. I totally agree that separating `Symbol.id` and `Symbol.info` makes sense, similarly how introduction of primary keys in databases makes sense.

In Dotty's case, the Denotation is the original abstraction, there are a lot of denotations that do not correspond to any specific symbols or definitions. This is a lot more than simple separation of a primary key that I see here.

I stand unconvinced and believe that the terms `Declaration` \ `Definiton` represent it better. Otherwise we can have a big long-term problem in terminology between Dotty and scalameta. As compilers and meta-programming are close to each-other I believe there is a huge value to keep terminology consistent.